### PR TITLE
Fix storagesync conflict behaviour

### PR DESCRIPTION
**Pull request by mateuszkrasucki**
FIX storageSync checksum conflict behaviour  …
- In case of checksum conflict "synced" file will always
  take precedence as the sync is one way so the "syncing" deployment
  is the one from which the file originates and in case previous
  attempt to sync was interrupted, it should be allowed to overwrite.

### CODE REVIEW

File: service/storage/storage_test.go [hunk 1]
```diff
@@ -919,32 +919,57 @@ func TestSyncFile(t *testing.T) {
...
+					s.EXPECT().Delete(gomock.Any(), "BUCKET", "FILE3", "V1").Return(fmt.Errorf("Error")),
 				}
 			},
 			nil,
 			withErrors,
 			nil,
 		},
 		{
-			"Write successfull",
+			"Already exists - conflict, successful delete and write",
```
Comment (dominikznidar) [inline service/storage/storage_test.go]: We are missing the successful delete, failed write option

File: storage/s3/s3.go [hunk 2]
```diff
@@ -79,10 +80,11 @@ type Minio interface {
 	ListBuckets() ([]minio.BucketInfo, error)
 	ListObjectsV2(bucketName, prefix string, recursive bool, doneCh <-chan struct{}) <-chan minio.ObjectInfo
 	GetObjectWithContext(ctx context.Context, bucketName, objectName string, opts minio.GetObjectOptions) (io.ReadCloser, error)
-	GetEncryptedObject(bucketName, objectName string, encryptMaterials encrypt.Materials) (io.ReadCloser, error)
+	GetEncryptedObject(bucketName, objectName string, encryptMateriadls encrypt.Materials) (io.ReadCloser, error)
```
Comment (dominikznidar) [inline storage/s3/s3.go]: small typo added to materials

File: service/storage/storage.go [hunk 3]
```diff
@@ -366,9 +362,12 @@ func (s *service) SyncFile(ctx context.Context, bucketID, fileID, version string
 		return fd, ErrAlreadyExists
 	// Already exists and conflicts
 	case err == nil && checksum != fd.Checksum:
-		s.logger.Error().
-			Msg("File already exists and has conflicting checksum")
-		return nil, ErrAlreadyExistsConflict
+		s.logger.Debug().
```
Comment (dominikznidar) [inline service/storage/storage.go]: Can we bump this message to `Info` so we can observe it on staging and production without all the debug noise?

**Comment by mateuszkrasucki** [comment]
@dominikznidar fixed

**Pull request merged**